docker/Dockerfile.mpich314_cuda101

```diff
@@ -0,0 +1,114 @@
+###
```

Comment:
Naming is a bit awkward. One is named `Dockerfile.mpich314_intel` the other is called `Dockerfile.mpich314_cuda101`. Do we need the MPI version in the name (or can we just settle on MPICH 3.14 for all Dockerfiles)? Why is `intel` and `cuda101` the same qualifier? Is `cuda101` actually `gcc_cuda101`?

Reply:
I suspect that we will have multiple MPI implementations in the near future.  For example, if we want GPUdirect communication on Google Cloud, then we would need a fv3gfs-fortran Docker image based on OpenMPI to enable G2G communication.  We are sticking to MPICH 3.1.4 currently because of its requirement to work on the Crays. 

As for a naming convention, I don't have a preference:  Dockerfile.gnu8_mpich314_cuda101, Dockerfile.intel2021_mpich314_nocuda ?

Reply:
I like your suggestion!

Reply:
Renamed configure files!